src/nodes/NodeManager.ts

```diff
@@ -385,11 +402,11 @@ class NodeManager {
   @ready(new nodesErrors.ErrorNodeManagerNotRunning())
   public async relayHolePunchMessage(message: nodesPB.Relay): Promise<void> {
     const conn = await this.getConnectionToNode(
-      message.getTargetId() as NodeId,
+      nodesUtils.decodeNodeId(message.getTargetId())!,
     );
     await conn.sendHolePunchMessage(
-      message.getSrcId() as NodeId,
-      message.getTargetId() as NodeId,
+      nodesUtils.decodeNodeId(message.getSrcId())!,
+      nodesUtils.decodeNodeId(message.getTargetId())!,
```

Comment:
Again, on a service boundary. Do we expect every hole punch message to have valid source and target node IDs? Or can these be malformed during transit? I'd assume the latter to be safer.